### ISSUE TITLE
IndexOutBoundException occurs when 500-1000 columns of data are given

### ISSUE THREAD

**Issue by fukasawah** [kamatama41/embulk-output-multi#22]
I encounted this problem when I gave it 500-1000 columns of data.

Full config, log and testdata here. [reports.zip](https://github.com/kamatama41/embulk-output-multi/files/4931395/reports.zip)

```
2020-07-16 21:02:50.229 +0900 [WARN] (0020:task-0000): Output for file_0 on index 0 failed.
java.lang.IndexOutOfBoundsException: end index (16875520) must not be greater than size (16777216)
        at io.airlift.slice.Preconditions.checkPositionIndexes(Preconditions.java:80) ~[na:na]
        at io.airlift.slice.Slice.<init>(Slice.java:126) ~[na:na]
        at io.airlift.slice.Slices.wrappedBuffer(Slices.java:139) ~[na:na]
        at org.embulk.deps.buffer.SliceImpl.<init>(SliceImpl.java:8) ~[na:na]
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[na:1.8.0_242]
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[na:1.8.0_242]
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[na:1.8.0_242]
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[na:1.8.0_242]
        at org.embulk.deps.buffer.Slice.createWithWrappedBuffer(Slice.java:12) ~[embulk.bat:0.9.23]
        at org.embulk.spi.PageReader.setPage(PageReader.java:38) ~[embulk.bat:0.9.23]
        at org.embulk.standards.CsvFormatterPlugin$1.add(CsvFormatterPlugin.java:116) ~[embulk.bat:0.9.23]
        at org.embulk.spi.FileOutputRunner$DelegateTransactionalPageOutput.add(FileOutputRunner.java:149) ~[embulk.bat:0.9.23]
        at org.embulk.output.multi.MultiTransactionalPageOutput$Delegate.add(MultiTransactionalPageOutput.java:111) ~[embulk-output-multi-0.4.1.jar:na]
        at org.embulk.output.multi.MultiTransactionalPageOutput.lambda$add$1(MultiTransactionalPageOutput.java:40) [embulk-output-multi-0.4.1.jar:na]
        at org.embulk.output.multi.MultiTransactionalPageOutput.applyToAllPlugins(MultiTransactionalPageOutput.java:72) [embulk-output-multi-0.4.1.jar:na]
        at org.embulk.output.multi.MultiTransactionalPageOutput.add(MultiTransactionalPageOutput.java:40) [embulk-output-multi-0.4.1.jar:na]
        at org.embulk.spi.PageBuilder.doFlush(PageBuilder.java:214) [embulk.bat:0.9.23]
        at org.embulk.spi.PageBuilder.flush(PageBuilder.java:219) [embulk.bat:0.9.23]
        at org.embulk.spi.PageBuilder.addRecord(PageBuilder.java:198) [embulk.bat:0.9.23]
        at org.embulk.standards.CsvParserPlugin.run(CsvParserPlugin.java:368) [embulk.bat:0.9.23]
        at org.embulk.spi.FileInputRunner.run(FileInputRunner.java:140) [embulk.bat:0.9.23]
        at org.embulk.spi.util.Executors.process(Executors.java:62) [embulk.bat:0.9.23]
        at org.embulk.spi.util.Executors.process(Executors.java:38) [embulk.bat:0.9.23]
        at org.embulk.exec.LocalExecutorPlugin$DirectExecutor$1.call(LocalExecutorPlugin.java:170) [embulk.bat:0.9.23]
        at org.embulk.exec.LocalExecutorPlugin$DirectExecutor$1.call(LocalExecutorPlugin.java:167) [embulk.bat:0.9.23]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [na:1.8.0_242]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [na:1.8.0_242]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [na:1.8.0_242]
        at java.lang.Thread.run(Thread.java:748) [na:1.8.0_242]
```

But, if not use embulk-output-multi, it works.

Also, I've found that it works well when `-X page_size=16MB` is given.
Although it seems to be proportional to the number of records and the number of columns, we don't know the detailed condition.

Workaround, split the configuration file and run it multiple times...

**Comment by kamatama41** [comment]
@fukasawah Thank you for reporting issue! (I didn't notice this until now 😓) I will take a look.

**Comment by reizist** [comment]
@kamatama41 Hi, I also encountered this issue, is there any updates? 😁

**Comment by kamatama41** [comment]
@fukasawah @reizist 
Sorry for very late response 😅 
It's fixed by `0.5.0`, please retry with the version, thanks.

**Comment by fukasawah** [comment]
It worked!
Thank you!!
